### PR TITLE
doc: Added instructions for how to add an upsteam to forked repo

### PR DESCRIPTION
**Pull request by dannmat**
As a first time git developer, I struggled to understand whether to create a new fork for each pull request or not.

After asking the IRC chat, I have added this to the documentation to further help new developers using git.

### CODE REVIEW

**Comment by kallewoof** [comment]
I'm not sure this belongs here. It looks like git/github documentation.

**Comment by dannmat** [comment]
I added it here because the instructions above it tell the user to 
1. fork
2. branch
3. commit

but it's not clear that the first step should only be taken once and the user should add an upstream, so I was adding clarification.

**Comment by kallewoof** [comment]
Maybe change the "1" to something like "1. Fork repository ([only the first time](https://help.github.com/en/articles/configuring-a-remote-for-a-fork))".

**Comment by laanwj** [comment]
> Maybe change the "1" to something like "1. Fork repository (only the first time)".

+1 on this, better to integrate these kind of remarks into the process

**Comment by dannmat** [comment]
Yes will look neater, I'll do this later on tonight.

**Comment by carnhofdaki** [comment]
ACK

This is about learning. When one learns something and someone else can learn from that, it improves us as a group. Welcome and all the best @dannmat !

@dannmat No need to change the patch but think of one more way to do it. More lazy than what is written in the linked GitHub's article. Hint: `origin`. Which repo is usually cloned first? (even before any change is done to the source code)

**Comment by dannmat** [comment]
Thanks! I'll have a read up on origin now!

**Comment by dannmat** [comment]
Changed and squashed

**Comment by carnhofdaki** [comment]
@dannmat Nice! Short and lean, though useful addition.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ facilitates social contribution, easy testing and peer review.
 
 To contribute a patch, the workflow is as follows:
 
-  1. Fork repository
+  1. Fork repository [only first time](https://help.github.com/en/articles/configuring-a-remote-for-a-fork).
```
Comment (fanquake) [inline CONTRIBUTING.md]: Can you change this to [kallewoof's suggestion](https://github.com/bitcoin/bitcoin/pull/17157#issuecomment-542567216) above.

```suggestion
  1. Fork repository ([only the first time](https://help.github.com/en/articles/configuring-a-remote-for-a-fork)).
```

**Comment by dannmat** [comment]
Rebased and squashed

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ facilitates social contribution, easy testing and peer review.
 
 To contribute a patch, the workflow is as follows:
 
-  1. Fork repository
+  1. Fork repository ([only the first time](https://help.github.com/en/articles/configuring-a-remote-for-a-fork)).
```
Comment (hebasto) [inline CONTRIBUTING.md]: Maybe link to https://help.github.com/en/articles/fork-a-repo is more relevant here?

**Review by hebasto — COMMENTED** [review]
Concept ACK.

File: CONTRIBUTING.md (hunk 2 above)
Comment (dannmat) [reply CONTRIBUTING.md]: Yes this is more relevant to the topic. I'll change this

Comment (carnhofdaki) [reply CONTRIBUTING.md]: What about one level closer to the source?
https://git-scm.com/book/en/v2/GitHub-Contributing-to-a-Project

Comment (carnhofdaki) [reply CONTRIBUTING.md]: Bit still, I don't get it. They mention `upstream` also in the Pro Git book.

What is wrong about this?

```
git clone https://github.com/bitcoin/bitcoin
cd bitcoin
git remote add myfork git@github.com:...
```

I.e. `origin` is `upstream` here and there is no need to write the whole word of `upstream` or change the `master` branch to be pulled from `upstream` remote...

Comment (carnhofdaki) [reply CONTRIBUTING.md]: Above is a request for comments. Thanks.

**Review by fanquake — APPROVED** [review]
ACK f09ba060cacd42e4cb9a242c1d731deb1f6623c6 - For such a simple change, I think we've bike-shed this enough already. The `bitcoin/bitcoin` repo isn't really where anyone should be learning how to use `git` etc, but I think linking out here is ok.

**Pull request merged**